Expensify/App
# Introduce Default Rooms into EChat

@robertjchen, please review when you have the chance
CC: @yuwenmemon 

### Details
This PR is part 1 of a set of PRs to include default rooms into echat. This PR sets them up to be searchable in the side bar and gives them their appropriate report name. There's also a termporary avatar to distinguish them from other regular chats. This also hides these chats behind a beta so that you can't accidentally access them in their incomplete state.

Another PR will add the correct avatar logic, as well as the logic to show the right reportName as the title.

This can be merged right after
- [x] https://github.com/Expensify/Expensify.cash/pull/3534 is merged.

### Fixed Issues

https://github.com/Expensify/Expensify/issues/161777

### Tests
Same as QA, but to verify that the beta is working correctly, you'll have to change [this line](https://github.com/Expensify/Expensify.cash/blob/6052430b8525161f92d0c014db94aeade1127330/src/libs/Permissions.js#L51) to `return false` so that `Permissions.canUseDefaultRooms(betas)` doesn't always return true since it's called in local dev.


### QA Steps
Feel free to reach out to Amal also, he can test this really quick.
1. Log into your expensify account in staging.expensify.cash.com
2. Verify you are NOT in the defaultRooms beta in https://www.expensify.com/_support/beta (you can check your betas in Onyx through Application -> Local Storage in js console)
3. Click on the search bar and verify you don't see any HUGE chatrooms (would contain all the expensify employees) with this placeholder image
4. Add yourself to the beta and wait a bit (might take 20 mins, might take an hour; you can check if you're in the beta by checking the onyx key in Application -> Local Storage in JS console).
5. Verify now that you see those previously filtered chat rooms in the search bar (see first screenshot below for comparrison).
6. Verify that when you click on it, the right title shows up at the top and (second screenshot below). Please do not type anything into them right now, it would be chaotic for users that are not in the beta.
7. On web/desktop, use the JS console to check Application -> Local Storage and filter by `report_`. You should see some reports have a non empty chatType (third screenshot below)


### Tested On


### Screenshots


#### Web
What a default chat room would like right now in the search bar, if you were in the beta (it's the `potato@ssdos.com` one): 
![image](https://user-images.githubusercontent.com/19364431/122623169-1527b780-d069-11eb-9954-dfe2d074778c.png)

![image](https://user-images.githubusercontent.com/19364431/122809729-585d7280-d283-11eb-9d88-2b24f52cc8ff.png)


![image](https://user-images.githubusercontent.com/19364431/122573852-061b1800-d01d-11eb-9885-5c717f8853a4.png)
